README.md

```diff
@@ -1 +1,145 @@
+
+All abovementioned methods support all parameters supported by `requests.request()` functions - as described in the [documentation](https://requests.readthedocs.io/en/latest/api/#main-interface).
+
+#### Initialization
+
+The core class is `keboola.client.http.HttpClient`, which can be initialized by specifying the `base_url` parameter:
+
+```python
+from keboola.client import HttpClient
+
+BASE_URL = 'https://connection.keboola.com/v2/storage/'
+cl = HttpClient(BASE_URL)
+```
+
+#### Default arguments
+
+For `HttpClient`, it is possible to define default arguments, which will be sent with every request. It's possible to define `default_http_header`, `auth_header` and `default_params` - a default header, a default authentication header and default parameters, respectively.
+
+```python
+from keboola.client import HttpClient
+
+BASE_URL = 'https://connection.keboola.com/v2/storage/'
+AUTH_HEADER = {
+    'x-storageapi-token': '1234-STORAGETOKENSTRING'
+}
+DEFAULT_PARAMS = {
+    'include': 'columns'
+}
+DEFAULT_HEADER = {
+    'Content-Type': 'application/json'
+}
+
+cl = HttpClient(BASE_URL, default_http_header=DEFAULT_HEADER,
+                auth_header=AUTH_HEADER, default_params=DEFAULT_PARAMS)
+```
+
+#### Basic authentication
+
+By specifying the `auth` argument, the `HttpClient` will utilize the basic authentication.
+
+```python
+from keboola.client import HttpClient
+
+BASE_URL = 'https://connection.keboola.com/v2/storage/'
+USERNAME = 'TestUser'
+PASSWORD = '@bcd1234'
+
+cl = HttpClient(BASE_URL, auth=(USERNAME, PASSWORD))
+```
+
+#### Simple POST request
+
+Making a simple POST request using `post_raw()` method.
+
+```python
+from keboola.client import HttpClient
+
+BASE_URL = 'https://www.example.com/change'
+cl = HttpClient(BASE_URL)
+
+data = {'attr_1': 'value_1', 'attr_2': 'value_2'}
+header = {'content-type': 'application/json'}
+response = cl.post_raw(self.base_url, data=data, headers=header)
+
+if response.ok is not True:
+    raise ValueError(response.json())
+else:
+    return response.json()
+```
+
+Making a simple POST request using `post()` method.
+
+```python
+from keboola.client import HttpClient
+
+BASE_URL = 'https://www.example.com/change'
+cl = HttpClient(BASE_URL)
+
+data = {'attr_1': 'value_1', 'attr_2': 'value_2'}
+header = {'content-type': 'application/json'}
+response = cl.post(self.base_url, data=data, headers=header)
+```
+
+#### Usage Example
+
+```python
+import os
+from urllib.parse import urljoin
+from keboola.client import HttpClient
+
+BASE_URL = 'https://connection.keboola.com/v2/'
+TOKEN = os.environ['TOKEN']
+
+cl = HttpClient(BASE_URL, auth_header={'x-storageapi-token': TOKEN})
+
+request_params = {'exclude': 'components'}
+response = cl.get_raw(url=urljoin(self.base_url, 'storage'),
```

Comment:
Now looking at this urljoin `get_raw(url=urljoin(self.base_url, 'storage')` I just realized it's kind of silly to do that every time you want to call some endpoint. What's the purpose of the base_url then right? What do you think about refactoring the API so that you can pass only the enpoint_path and the concat would happen inside the method itself?

So the signature would be something like `get_raw(endpoint_path, is_absolute_path=False, ...):` where the `is_absolute_path` would allow you to override the default behaviour and provide full URL on the first argument?


Which brings me to another thing I've noticed and that's signatures of the functions. It drove me crazy before that I don't know what is the required parameters (for instance the [post method](https://github.com/keboola/python-http-client/blob/f8e147e4a6f37118f59f4d3a487d4f3abd7438d6/src/keboola/client/http.py#L140)  `def post(self, *args, ignore_auth=False, **kwargs)`  It's always better to be explicit

I think the signature should at least be the similar to the sessions.request method (without method):

```
request(self, method, url,
            params=None, data=None, headers=None, cookies=None, files=None,
            auth=None, timeout=None, allow_redirects=True, proxies=None,
            hooks=None, stream=None, verify=None, cert=None, json=None)
```

perhaps just:

```
post(self, url,
            params=None, data=None, headers=None, cookies=None, files=None,
            json=None, **kwargs)
```
So the less common parameters are hidden.

I took this from Keboola Storage client package and was too lazy to refactor it but it bothers me every time I'm using it, looking up what's the parameter names.